boa_engine/src/bytecompiler/mod.rs

```diff
@@ -1239,28 +1245,36 @@ impl<'b, 'host> ByteCompiler<'b, 'host> {
             New,
         }
 
-        let (call, kind) = match callable {
-            Callable::Call(call) => match call.function() {
-                Expression::Identifier(ident) if *ident == Sym::EVAL => (call, CallKind::CallEval),
-                _ => (call, CallKind::Call),
-            },
+        let (call, mut kind) = match callable {
+            Callable::Call(call) => (call, CallKind::Call),
             Callable::New(new) => (new.call(), CallKind::New),
         };
 
-        match call.function() {
+        let mut function = call.function();
+        while let Expression::Parenthesized(p) = function {
+            function = p.expression();
+        }
```

Comment:
Saw this pattern on several places. Maybe it should be extracted to something like `Expression::flatten`